invenio_config/utils.py

```diff
@@ -77,3 +80,60 @@ def create_conf_loader(*args, **kwargs):  # pragma: no cover
         DeprecationWarning,
     )
     return create_config_loader(*args, **kwargs)
+
+
+def build_db_uri():
+    """Build the database URI."""
+    default_db_uri = "postgresql+psycopg2://invenio-app-rdm:invenio-app-rdm@localhost/invenio-app-rdm"
+    params = {
+        k: os.environ.get(f"DB_{k.upper()}")
+        for k in ["user", "password", "host", "port", "name"]
+    }
+
+    if all(params.values()):
+        uri = f"postgresql+psycopg2://{params['user']}:{params['password']}@{params['host']}:{params['port']}/{params['name']}"
+        return uri
+
+    uri = os.environ.get("SQLALCHEMY_DATABASE_URI")
+    if uri:
+        return uri
+    return default_db_uri
+
+
+def build_broker_url():
```

Comment:
Do you think we could add the protocol and vhost? https://github.com/inveniosoftware/helm-invenio/commit/728173dcf2f58d851da22fc365bd0b05bcdba008

Reply:
Protocol has been made required, while vhost is optional in the updated implementation. Both are supported now. I also added the INVENIO_ prefix as the highest priority, falling back to defaults if not set.